src/App.js

```diff
@@ -1,23 +1,123 @@
-
+import React, { useState, useEffect } from "react"
+import List from "./Components/List"
+import {nanoid} from "nanoid"
 import './App.css';
+import { AiOutlineSearch } from "react-icons/ai"
 
 function App() {
+  const [friendsList, setFriendsList] = useState([])
+  const [newFriend, setNewFriend] = useState("")
+  const [filteredFriends, setFilteredFriends] = useState([])
+
+function handleChange(event) {
+  const { value } = event.target;
+  setNewFriend(value)
+}
+
+function handleSubmit(event) {
+  event.preventDefault()
+  const friendToBeAdded = {
+    name: newFriend,
+    isFavourite: false,
+    id: nanoid()
+  } 
+    setFriendsList([
+      ...friendsList,
+      friendToBeAdded
+    ])
+    setNewFriend("")    
+}
+
+function toggleFavourites(id) {
+  setFriendsList(prevArr => prevArr.map(star=> {
+    return star.id === id ?
```

Comment:
extract this login out of setFriendsList

Reply:
didn't get u.. did u mean i should write like...
function toggleFavourites(id) {
  setFriendsList( friendsList.map( star => {
    return star.id === id ?
    {...star, isFavourite: !star.isFavourite} : star
  }))
}